File: csharp/main/ImportTypeDef.cs
using System;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public class ImportTypeDef : ImportDef
	{
		public ImportTypeDef([NotNull] TypeCustom target, [CanBeNull] String alias = null)
			: base(ImportKind.Type)
		{
			Target = target;
			Alias = alias;
		}

		[NotNull]
		public TypeCustom Target { get; }

		[CanBeNull]
		public String Alias { get; }
	}
}

File: java/src/main/java/deltix/luminary/ImportTypeDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ImportTypeDef extends ImportDef {
    private TypeCustom target;
    private String alias;

    protected ImportTypeDef(@NotNull TypeCustom target, @Nullable String alias) {
        super(ImportKind.TYPE);
        this.target = target;
        this.alias = alias;
    }

    @NotNull
    public TypeCustom getTarget() {
        return target;
    }

    @Nullable
    public String getAlias() {
        return alias;
    }
}
